### PR TITLE
PHPstan level 6 for Migrator.php

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- PHPstan level 6 for Migrator.php

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

**Comment by Alkarex** [comment]
Don't you mean Level 6?

File: lib/Minz/Migrator.php [hunk 1]
```diff
@@ -21,16 +21,16 @@ class Minz_Migrator
 	 * @param string $migrations_path
 	 * @param string $applied_migrations_path
 	 *
-	 * @throws BadFunctionCallException if a callback isn't callable.
-	 * @throws DomainException if there is no migrations corresponding to the
+	 * @return boolean|string Returns true if execute succeeds to apply
```
Comment (Alkarex) [inline lib/Minz/Migrator.php]:
```suggestion
	 * @return true|string Returns true if execute succeeds to apply
```

File: lib/Minz/Migrator.php [hunk 2]
```diff
@@ -21,16 +21,16 @@ class Minz_Migrator
...
+	 *                        migrations, or a string if it fails.
+	 *@throws DomainException if there is no migrations corresponding to the
```
Comment (Alkarex) [inline lib/Minz/Migrator.php]:
```suggestion
	 * @throws DomainException if there is no migrations corresponding to the
```

File: lib/Minz/Migrator.php [hunk 3]
```diff
@@ -55,7 +55,7 @@ public static function execute($migrations_path, $applied_migrations_path) {
 		}
 
 		$lock_path = $applied_migrations_path . '.lock';
-		if (!@mkdir($lock_path, 0770, true)) {
+		if (!mkdir($lock_path, 0770, true) && !is_dir($lock_path)) {
```
Comment (Alkarex) [inline lib/Minz/Migrator.php]: Why did you remove the `@`? It risks breaking some outputs

File: lib/Minz/Migrator.php [hunk 4]
```diff
@@ -180,9 +178,9 @@ public function addMigration($version, $callback) {
 	 *
 	 * @see https://www.php.net/manual/en/function.strnatcmp.php
 	 *
-	 * @return array
+	 * @return array<string, callable>
```
Comment (Alkarex) [inline lib/Minz/Migrator.php]:
```suggestion
	 * @return array<string,callable>
```

My syntax highlight does not work well when there are spaces in types :-)

File: lib/Minz/Migrator.php (hunk 3 above)
Comment (Alkarex) [reply lib/Minz/Migrator.php]: Why the `!is_dir($lock_path)` ? This looks wrong

**Comment by Alkarex** [comment]
Contributes to https://github.com/FreshRSS/FreshRSS/issues/4112

File: lib/Minz/Migrator.php (hunk 3 above)
Comment (ColonelMoutarde) [reply lib/Minz/Migrator.php]: that's for remove the @

**Comment by ColonelMoutarde** [comment]
> Don't you mean Level 6?

you'r right sorry

File: lib/Minz/Migrator.php (hunk 3 above)
Comment (Alkarex) [reply lib/Minz/Migrator.php]: But I do not believe your new version is equivalent at all. The first mkdir is supposed to fail if the path exists. It is a mutex

File: lib/Minz/Migrator.php [hunk 5]
```diff
@@ -247,11 +245,11 @@ public function upToDate() {
 	 * considered as successful. It is considered as good practice to return
 	 * true on success though.
 	 *
-	 * @return array Return the results of each executed migration. If an
+	 * @return array<string|bool> Return the results of each executed migration. If an
```
Comment (Alkarex) [inline lib/Minz/Migrator.php:248]: I believe this might be the following (to be double-checked)

```suggestion
	 * @return array<string|true> Return the results of each executed migration. If an
```

File: lib/Minz/Migrator.php (hunk 3 above)
Comment (Alkarex) [reply lib/Minz/Migrator.php]: I am reverting this line, as i believe the new one is wrong

```suggestion
		if (!@mkdir($lock_path, 0770, true)) {
```
